clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -706,8 +706,17 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
     // 1 - disable unroll.
     // other positive integer n - unroll by n.
     if (OpenCLHint || UnrollHint) {
-      ValueInt = OpenCLHint ? OpenCLHint->getUnrollHint()
-                            : UnrollHint->getUnrollHint();
+      if (OpenCLHint) {
+        ValueInt = OpenCLHint->getUnrollHint();
+      } else { // UnrollHint
+        if (auto *ValueExpr = UnrollHint->getUnrollHintExpr()) {
+          llvm::APSInt ValueAPS = ValueExpr->EvaluateKnownConstInt(Ctx);
+          ValueInt = ValueAPS.getSExtValue();
+        } else {
+          ValueInt = 0;
+        }
+      }
```

Comment:
Maybe something like:
```
if (OpenCLHint) {
  ValueInt = OpenCLHint->getUnrollHint();
}

if (UnrollHint) {
  // code for UnrollHint
}
```

to reduce nesting?

Reply:
It actually doesn't reduce nesting, if we are talking about lines 709-718.
I tried to split the check on line 708, but it led to duplication of state and option setting on lines 720-725

Reply:
Oh, sorry for the obscure suggestion. I aimed to highlight the entire condition removal at line 708.

Reply:
I've missed that there's more shared code between the two. Should you find an appropriate way to reduce nesting, this would be welcome, but otherwise, I'm fine with this approach.